<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Run pipeline - exseek</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Run pipeline";
    var mkdocs_page_input_path = "run_pipeline.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> exseek</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Overview</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../get_started/">Get Started</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../installation/">Installation</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../genome_and_annotatins.md">Genome and Annotations</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../small_rna_mapping/">Small RNA-seq mapping</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../long_rna_mapping/">Long RNA-seq mapping</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../matrix_processing/">Matrix processing</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../feature_selection.md">Feature Selection</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../cluster_configuration/">Run on a cluster</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../configuration/">Configuration File Reference</a>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Utilities</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../quality_control/">Quality Control</a>
                </li>
                <li class="">
                    
    <a class="" href="../adapter_removal/">Adapter Removal</a>
                </li>
                <li class="">
                    
    <a class="" href="../bigwig/">Generate BigWig Files</a>
                </li>
                <li class="">
                    
    <a class="" href="../igv_browser/">IGV Browser</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../FAQ/">Frequently Asked Questions</a>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">exseek</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>Run pipeline</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h2 id="generate-sequential-mapping-snakefile">Generate sequential mapping snakefile</h2>
<pre><code class="bash">snakemake --snakefile snakemake/prepare_genome.snakemake --configfile snakemake/config.yaml --rerun-incomplete -k
</code></pre>

<h2 id="scirep">SciRep</h2>
<pre><code class="bash">
bin/generate_snakemake.py sequential_mapping --rna-types rRNA,miRNA,piRNA,Y_RNA,srpRNA,tRNA,snRNA,snoRNA,lncRNA,mRNA,tucpRNA \
    -o snakemake/sequential_mapping.snakemake
snakemake --snakefile snakemake/mapping_small.snakemake --configfile config/scirep.yaml --rerun-incomplete -k \
    --cluster-config snakemake/cluster.yaml \
    --cluster &quot;bsub -q {cluster.queue} -J {cluster.name} -e {cluster.stderr} -o {cluster.stdout} -R {cluster.resources} -n {cluster.threads}&quot;


snakemake --snakefile snakemake/expression_matrix.snakemake --configfile config/scirep.yaml --rerun-incomplete -k

snakemake --snakefile snakemake/feature_selection.snakemake --configfile config/scirep.yaml --rerun-incomplete -k

snakemake --snakefile snakemake/feature_selection.snakemake --configfile config/scirep.yaml --rerun-incomplete -k \
    --cluster-config snakemake/cluster.yaml \
    --cluster &quot;bsub -q {cluster.queue} -J {cluster.name} -e {cluster.stderr} -o {cluster.stdout} -R {cluster.resources} -n {cluster.threads}&quot;

snakemake --snakefile snakemake/call_domains_long.snakemake --configfile config/scirep.yaml --rerun-incomplete -k \
    --cluster-config snakemake/cluster.yaml \
    --cluster &quot;bsub -q {cluster.queue} -J {cluster.name} -e {cluster.stderr} -o {cluster.stdout} -R {cluster.resources} -n {cluster.threads}&quot;
snakemake --snakefile snakemake/bigwig.snakemake --configfile config/scirep.yaml --rerun-incomplete -k \
    --cluster-config snakemake/cluster.yaml \
    --cluster &quot;bsub -q {cluster.queue} -J {cluster.name} -e {cluster.stderr} -o {cluster.stdout} -R {cluster.resources} -n {cluster.threads}&quot;

export PATH=$PWD/singularity/wrappers:$PATH
PATH=&quot;$PWD/singularity/wrappers:$PATH&quot; snakemake --snakefile snakemake/normalization.snakemake --configfile config/scirep.yaml --rerun-incomplete -k

bin/report.py visualize_domains --sample-ids-file data/scirep/sample_ids.txt \
    --output-dir output/scirep \
    --count-matrix output/scirep/count_matrix/domains_combined.txt \
    --features output/scirep/feature_selection/filter.scimpute_count.Norm_CPM.Batch_RUV.domains_combined/Normal-CRC/random_forest.10.robust/features.txt \
    --chrom-sizes genome/hg38/chrom_sizes/transcriptome_genome \
    --output-file tmp/visualize_domains.pdf
bin/feature_selection.py calculate_clustering_score \
    --matrix output/scirep/matrix_processing/filter.scimpute_count.Norm_null.domains_combined.txt \
    --sample-classes data/scirep/sample_classes.txt --transpose

snakemake --snakefile snakemake/evaluate_features.snakemake --configfile config/scirep.yaml
</code></pre>

<h2 id="lulab-hcc">Lulab HCC</h2>
<pre><code class="bash">/Share/home/caojingyi/exRNA/process/18.new_hcc_lulab/Snakefile
snakemake --snakefile snakemake/mapping_small.snakemake --configfile config/lulab_hcc.yaml --rerun-incomplete -k \
    --cluster-config snakemake/cluster.yaml \
    --cluster &quot;bsub -q {cluster.queue} -J {cluster.name} -e {cluster.stderr} -o {cluster.stdout} -R {cluster.resources} -n {cluster.threads}&quot; \

snakemake --snakefile snakemake/mapping_small.snakemake --configfile config/lulab_hcc.yaml --rerun-incomplete -k
snakemake --snakefile snakemake/expression_matrix.snakemake --configfile config/lulab_hcc.yaml --rerun-incomplete -k
snakemake --snakefile snakemake/normalization.snakemake --configfile config/lulab_hcc.yaml --rerun-incomplete -k
snakemake --snakefile snakemake/feature_selection.snakemake --configfile config/lulab_hcc.yaml --rerun-incomplete -k \
    --cluster-config snakemake/cluster.yaml \
    --cluster &quot;bsub -q {cluster.queue} -J {cluster.name} -e {cluster.stderr} -o {cluster.stdout} -R {cluster.resources} -n {cluster.threads}&quot;

snakemake --snakefile snakemake/evaluate_features.snakemake --configfile config/lulab_hcc.yaml
</code></pre>

<h2 id="exorbase">exoRBase</h2>
<pre><code class="bash"># subsample reads
[ -d &quot;data/exorbase_test/fastq&quot; ] || mkdir -p &quot;data/exorbase_test/fastq&quot;
cp data/exorbase/*.txt data/exorbase_test
for f in data/exorbase/fastq/*.fastq;do
    sample_id=$(basename $f)
    sample_id=${sample_id/.fastq/}
    echo data/exorbase_test/fastq/${sample_id}.fastq
    head -n 400000 $f &gt; data/exorbase_test/fastq/${sample_id}.fastq
done

snakemake --snakefile snakemake/mapping_long.snakemake --rerun-incomplete -k --configfile config/exorbase.yaml \
    --cluster-config snakemake/cluster.yaml \
    --cluster &quot;bsub -q {cluster.queue} -J {cluster.name} -e {cluster.stderr} -o {cluster.stdout} -R {cluster.resources} -n {cluster.threads}&quot; -j60
snakemake --snakefile snakemake/normalization.snakemake --configfile config/exorbase.yaml --rerun-incomplete -k
snakemake --snakefile snakemake/feature_selection.snakemake --configfile config/exorbase.yaml --rerun-incomplete -k

</code></pre>

<h2 id="tcga-mirna-seq-crc">TCGA miRNA-seq (CRC)</h2>
<pre><code class="bash">snakemake --snakefile snakemake/bam_to_fastx.snakemake --rerun-incomplete -k --configfile config/tcga_crc.yaml 
snakemake --snakefile snakemake/mapping_small.snakemake --rerun-incomplete -k --configfile config/tcga_crc.yaml \
    --cluster-config snakemake/cluster.yaml \
    --cluster &quot;bsub -q {cluster.queue} -J {cluster.name} -e {cluster.stderr} -o {cluster.stdout} -R {cluster.resources} -n {cluster.threads}&quot;
</code></pre>

<h2 id="tcga-mirna-seq-hcc">TCGA miRNA-seq (HCC)</h2>
<pre><code class="bash">snakemake --snakefile snakemake/bam_to_fastx.snakemake --rerun-incomplete -k --configfile config/tcga_hcc.yaml 
snakemake --snakefile snakemake/mapping_small.snakemake --rerun-incomplete -k --configfile config/tcga_hcc.yaml \
    --cluster-config snakemake/cluster.yaml \
    --cluster &quot;bsub -q {cluster.queue} -J {cluster.name} -e {cluster.stderr} -o {cluster.stdout} -R {cluster.resources} -n {cluster.threads}&quot;
snakemake --snakefile snakemake/bigwig.snakemake --configfile config/tcga_hcc.yaml --rerun-incomplete -k \
    --cluster-config snakemake/cluster.yaml \
    --cluster &quot;bsub -q {cluster.queue} -J {cluster.name} -e {cluster.stderr} -o {cluster.stdout} -R {cluster.resources} -n {cluster.threads}&quot;
snakemake --snakefile snakemake/call_domains_long.snakemake --configfile config/tcga_hcc.yaml --rerun-incomplete -k \
    --cluster-config snakemake/cluster.yaml \
    --cluster &quot;bsub -q {cluster.queue} -J {cluster.name} -e {cluster.stderr} -o {cluster.stdout} -R {cluster.resources} -n {cluster.threads}&quot;
</code></pre>

<h2 id="gse113994-healthy-cfrna-pnas-2018">GSE113994 (Healthy, cfRNA, PNAS 2018)</h2>
<pre><code class="bash">snakemake --snakefile snakemake/quality_control.snakemake --rerun-incomplete -k --configfile config/GSE113994.yaml \
    --cluster-config snakemake/cluster.yaml \
    --cluster &quot;bsub -q {cluster.queue} -J {cluster.name} -e {cluster.stderr} -o {cluster.stdout} -R {cluster.resources} -n {cluster.threads}&quot;
snakemake --snakefile snakemake/mapping_small.snakemake --rerun-incomplete -k --configfile config/GSE113994.yaml \
    --cluster-config snakemake/cluster.yaml \
    --cluster &quot;bsub -q {cluster.queue} -J {cluster.name} -e {cluster.stderr} -o {cluster.stdout} -R {cluster.resources} -n {cluster.threads}&quot;
</code></pre>

<h2 id="gse45722-healthy-exosome-bmc-genomics-2013">GSE45722 (Healthy, exosome, BMC Genomics 2013)</h2>
<pre><code class="bash">snakemake --snakefile snakemake/quality_control.snakemake --rerun-incomplete -k --configfile config/GSE45722.yaml \
    --cluster-config snakemake/cluster.yaml \
    --cluster &quot;bsub -q {cluster.queue} -J {cluster.name} -e {cluster.stderr} -o {cluster.stdout} -R {cluster.resources} -n {cluster.threads}&quot; -j40
snakemake --snakefile snakemake/mapping_small.snakemake --rerun-incomplete -k --configfile config/GSE45722.yaml \
    --cluster-config snakemake/cluster.yaml \
    --cluster &quot;bsub -q {cluster.queue} -J {cluster.name} -e {cluster.stderr} -o {cluster.stdout} -R {cluster.resources} -n {cluster.threads}&quot; -j40
</code></pre>

<h2 id="gse114711-healthy-exosome-scientific-reports-2018">GSE114711 (Healthy, exosome, Scientific Reports 2018)</h2>
<pre><code class="bash">snakemake --snakefile snakemake/quality_control.snakemake --rerun-incomplete -k --configfile config/GSE114711.yaml \
    --cluster-config snakemake/cluster.yaml \
    --cluster &quot;bsub -q {cluster.queue} -J {cluster.name} -e {cluster.stderr} -o {cluster.stdout} -R {cluster.resources} -n {cluster.threads}&quot; -j40
snakemake --snakefile snakemake/mapping_small.snakemake --rerun-incomplete -k --configfile config/GSE114711.yaml \
    --cluster-config snakemake/cluster.yaml \
    --cluster &quot;bsub -q {cluster.queue} -J {cluster.name} -e {cluster.stderr} -o {cluster.stdout} -R {cluster.resources} -n {cluster.threads}&quot; -j40
</code></pre>

<h2 id="gini-index-comparison-between-exrna-and-tcga">Gini index comparison between exRNA and TCGA</h2>
<pre><code class="bash">
for dataset in 'scirep' 'tcga_crc' 'tcga_hcc' 'GSE114711' 'GSE45722';do
    mkdir -p &quot;output/${dataset}/analysis&quot;
    echo bin/extract_bigwig.py abundant_rna_coverage_matrix \
        --matrix output/${dataset}/count_matrix/transcript.txt \
        --bigwig-pattern &quot;output/${dataset}/tbigwig/{sample_id}.{gene_type}.bigWig&quot; -n 100 \
        -o output/${dataset}/analysis/abundant_rna_coverage_matrix.h5
done
</code></pre>

<h2 id="integrated-dataset-scirep2016-gse45722-gse114711">Integrated dataset SciRep2016, GSE45722, GSE114711</h2>
<h3 id="copy-files">Copy files</h3>
<pre><code class="bash">dataset='exosome_small'
mkdir -p data/${dataset}
sources=(scirep GSE45722 GSE114711)
# sample_ids.txt
cat data/scirep/sample_ids.txt data/GSE45722/sample_ids.txt \
    data/GSE114711/sample_ids.txt &gt; data/${dataset}/sample_ids.txt
# sample_classes.txt
{
    awk 'BEGIN{OFS=&quot;\t&quot;;FS=&quot;\t&quot;}{print $1,$2}' data/scirep/sample_classes.txt 
    awk 'BEGIN{OFS=&quot;\t&quot;;FS=&quot;\t&quot;}{print $1,&quot;Healthy Control&quot;}' data/GSE45722/sample_classes.txt
    awk 'BEGIN{OFS=&quot;\t&quot;;FS=&quot;\t&quot;}{print $1,&quot;Healthy Control&quot;}' data/GSE114711/sample_classes.txt 
} &gt; data/${dataset}/sample_classes.txt
# batch_info.txt
{
    echo -e 'sample_id\tpublication'
    awk 'BEGIN{OFS=&quot;\t&quot;;FS=&quot;\t&quot;}{print $1,&quot;scirep&quot;}' data/scirep/sample_classes.txt 
    awk 'BEGIN{OFS=&quot;\t&quot;;FS=&quot;\t&quot;}{print $1,&quot;GSE45722_healthy&quot;}' data/GSE45722/sample_classes.txt
    awk 'BEGIN{OFS=&quot;\t&quot;;FS=&quot;\t&quot;}{print $1,&quot;GSE114711_healthy&quot;}' data/GSE114711/sample_classes.txt 
} &gt; data/exosome_small/batch_info.txt

# count_matrix/domains_long.txt
mkdir -p output/${dataset}/count_matrix
bin/preprocess.py merge_data_frames \
    -i output/scirep/count_matrix/transcript.txt \
    -i output/GSE45722/count_matrix/transcript.txt \
    -i output/GSE114711/count_matrix/transcript.txt \
    --on feature --fillna 0 \
    -o output/${dataset}/count_matrix/transcript.txt
# domains_by_sample/20/05
mkdir -p output/${dataset}/domains_by_sample/20/05
rsync -ra output/scirep/domains_by_sample/20/05/*.bed output/${dataset}/domains_by_sample/20/05
rsync -ra output/GSE45722/domains_by_sample/20/05/*.bed output/${dataset}/domains_by_sample/20/05
rsync -ra output/GSE114711/domains_by_sample/20/05/*.bed output/${dataset}/domains_by_sample/20/05
# tbed
mkdir -p output/${dataset}/tbed
mkdir -p output/${dataset}/gbed

for d in ${sources[@]};do
    rsync -rav output/$d/tbed/ output/${dataset}/tbed
    rsync -rav output/$d/gbed/ output/${dataset}/gbed
done


</code></pre>

<h3 id="analysis">Analysis</h3>
<pre><code class="bash">bin/exseek.py call_domains -d exosome_small
</code></pre>

<h2 id="data">Data</h2>
<h3 id="exorbase_1">exoRBase</h3>
<pre><code class="bash">[ -d data/exorbase/fastq ] || mkdir -p data/exorbase/fastq
ln -f -s /BioII/lulab_b/shared/projects/exRNA/published_exRNA/exosome_exoRBase/exosome_GSE100063_CRC/fastq/*.fastq data/exorbase/fastq
ln -f -s /BioII/lulab_b/caojingyi/exoRBase/fastq/HCC/*.fastq data/exorbase/fastq
ln -f -s /BioII/lulab_b/caojingyi/exoRBase/fastq/Normal/*.fastq  data/exorbase/fastq
ln -f -s /BioII/lulab_b/shared/projects/exRNA/published_exRNA/exosome_exoRBase/exosome_GSE100232_PAAD/fastq/*.fastq data/exorbase/fastq
ln -f -s /BioII/lulab_b/shared/projects/exRNA/published_exRNA/exosome_exoRBase/exosome_GSE99985_CHD/fastq/*.fastq data/exorbase/fastq
ln -f -s /BioII/lulab_b/caojingyi/exoRBase/fastq/*.fastq data/exorbase/fastq
ls data/exorbase/fastq | cut -d'_' -f1 | sort | uniq &gt; data/exorbase/sample_ids.txt
ln -f -s /BioII/zhuyumin/exLocator/exosome_SR2018_GSE114711/1.fastq/*.fastq data/GSE114711/fastq
</code></pre>

<h2 id="spike-in">Spike-in</h2>
<h3 id="small-rna-seq">Small RNA-seq</h3>
<p><code>/BioII/lulab_b/wangsiqi/exRNA/exRNA-panel/NEB/03.1811_T4PNK/ExiSEQ-spikeIn</code></p>
<pre><code class="bash">cp /BioII/lulab_b/wangsiqi/exRNA/exRNA-panel/NEB/03.1811_T4PNK/ExiSEQ-spikeIn/ExiSEQ-spikeIn.fa genome/hg38/fasta/spikein_small.fa
samtools faidx genome/hg38/fasta/spikein_small.fa
</code></pre>

<h3 id="long-rna-seq">Long RNA-seq</h3>
<p><code>/BioII/lulab_b/wangsiqi/exRNA/exRNA-panel/pico-smart/exSeek/ERCC-spikeIn</code></p>
<pre><code class="bash">cp /BioII/lulab_b/wangsiqi/exRNA/exRNA-panel/pico-smart/exSeek/ERCC-spikeIn/ERCC92.fa genome/hg38/fasta/spikein_long.fa
samtools faidx genome/hg38/fasta/spikein_long.fa
</code></pre>

<h2 id="bigwig-files">BigWig files</h2>
<pre><code class="bash">dest=&quot;/BioII/lulab_b/shared/projects/exSeek/output/exorbase/bigwig&quot;
[ -d &quot;$dest&quot; ] || mkdir -p &quot;$dest&quot;
rsync -rav --delete /Share/home/shibinbin/projects/exSeek-dev/output/exorbase/bigwig/ &quot;$dest/&quot;
</code></pre>

<pre><code class="bash">bin/create_igv.py -r genome/hg38 -g hg38 -i templates/igv/main.html \
     -o igv.html \
    --track output/exorbase/bigwig/SRR5679904.genome_rmdup.-.bigWig \
    --track output/exorbase/bigwig/SRR5679904.genome_rmdup.+.bigWig \
    --track output/exorbase/bigwig/SRR5679905.genome_rmdup.-.bigWig \
    --track output/exorbase/bigwig/SRR5679905.genome_rmdup.+.bigWig \
    --track output/exorbase/bigwig/SRR5679906.genome_rmdup.-.bigWig \
    --track output/exorbase/bigwig/SRR5679906.genome_rmdup.+.bigWig \
    --track output/exorbase/bigwig/SRR5679907.genome_rmdup.-.bigWig \
    --track output/exorbase/bigwig/SRR5679907.genome_rmdup.+.bigWig \
    --track output/exorbase/bigwig/SRR5679908.genome_rmdup.-.bigWig \
    --track output/exorbase/bigwig/SRR5679908.genome_rmdup.+.bigWig \
    --track output/exorbase/bigwig/SRR5679909.genome_rmdup.-.bigWig \
    --track output/exorbase/bigwig/SRR5679909.genome_rmdup.+.bigWig \
    --track output/exorbase/bigwig/SRR5687235.genome_rmdup.-.bigWig \
    --track output/exorbase/bigwig/SRR5687235.genome_rmdup.+.bigWig \
    --track output/exorbase/bigwig/SRR5687236.genome_rmdup.-.bigWig \
    --track output/exorbase/bigwig/SRR5687236.genome_rmdup.+.bigWig \
    --track output/exorbase/bigwig/SRR5687237.genome_rmdup.-.bigWig \
    --track output/exorbase/bigwig/SRR5687237.genome_rmdup.+.bigWig \
    --track output/exorbase/bigwig/SRR5687238.genome_rmdup.-.bigWig \
    --track output/exorbase/bigwig/SRR5687238.genome_rmdup.+.bigWig
</code></pre>

<h2 id="visualize-domains">Visualize domains</h2>
<pre><code class="bash">bin/report.py visualize_domains --sample-classes data/scirep/sample_classes.txt \
    --features output/scirep/feature_selection/filter.viper_count.Norm_CPM.Batch_RUV.domains_combined/Normal-CRC/random_forest.10.robust/features.txt \
    --genome-dir genome/hg38 \
    --flanking 20 -o tmp/visualize_domains.pdf --output-dir output/scirep
</code></pre>

<pre><code class="bash">[ -d &quot;output/scirep/visualize_domains&quot; ] || mkdir -p output/scirep/visualize_domains
for features in $(cat /home/chenxupeng/projects/exseek/output/selected_feature/scirep/selected_features.txt);do
    compare_group=$(echo $features | cut -d'/' -f5 )
    bin/report.py visualize_domains --sample-classes data/scirep/sample_classes.txt \
        --features $features \
        --genome-dir genome/hg38 \
        --flanking 20 -o output/scirep/visualize_domains/${compare_group}.pdf --output-dir output/scirep
done
</code></pre>

<h2 id="feature-selection">Feature selection</h2>
<pre><code class="bash">bin/machine_learning.py  cross_validation \
    --matrix output/scirep/matrix_processing/filter.viper_count.Norm_CPM.Batch_RUV.domains_combined.txt \
    --sample-classes data/scirep/sample_classes.txt \
    -o tmp/cross_validation \
    --transpose \
    --positive-class 'Colorectal Cancer' --negative-class 'Healthy Control' \
    --cv-params '{&quot;splitter&quot;: &quot;stratified_shuffle_split&quot;, &quot;n_splits&quot;: 5, &quot;test_size&quot;: 0.2}' \
    --zero-fraction-filter \
    --log-transform '{&quot;base&quot;: 2}' \
    --rpkm-filter '{&quot;threshold&quot;: 10}' \
    --scaler robust \
    --selector robust --selector-params '{&quot;cv&quot;: {&quot;splitter&quot;: &quot;stratified_shuffle_split&quot;, &quot;n_splits&quot;: 10, &quot;test_size&quot;: 0.2}}' \
    --grid-search \
    --grid-search-param-grid '{&quot;n_estimators&quot;: [25, 50, 75], &quot;max_depth&quot;: [3, 4, 5]}' \
    --grid-search-cv-params '{&quot;splitter&quot;: &quot;stratified_shuffle_split&quot;, &quot;n_splits&quot;: 5, &quot;test_size&quot;: 0.2}' \
    --classifier random_forest
</code></pre>

<h2 id="notes">Notes</h2>
<h3 id="20181229">2018.12.29</h3>
<ul>
<li>Feature selection with pairs of features</li>
<li>Find enriched rRNA fragments</li>
<li>Validate tRNA fragment features</li>
</ul>
              
            </div>
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>

</body>
</html>
