<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Prepare Genome and Annotation - exseek</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Prepare Genome and Annotation";
    var mkdocs_page_input_path = "prepare_genome_annotation.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> exseek</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Overview</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../get_started/">Get Started</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../installation/">Installation</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../genome_and_annotations/">Genome and Annotations</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../small_rna_mapping/">Small RNA-seq mapping</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../long_rna_mapping/">Long RNA-seq mapping</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../matrix_processing/">Matrix processing</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../feature_selection.md">Feature Selection</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../cluster_configuration/">Run on a cluster</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../configuration/">Configuration File Reference</a>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Utilities</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../quality_control/">Quality Control</a>
                </li>
                <li class="">
                    
    <a class="" href="../adapter_removal/">Adapter Removal</a>
                </li>
                <li class="">
                    
    <a class="" href="../bigwig/">Generate BigWig Files</a>
                </li>
                <li class="">
                    
    <a class="" href="../igv_browser/">IGV Browser</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../FAQ/">Frequently Asked Questions</a>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">exseek</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>Prepare Genome and Annotation</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="prepare-genome-and-annotation">Prepare Genome and Annotation</h1>
<ul>
<li>Sequence</li>
<li>Annotation</li>
<li>Index files</li>
</ul>
<h2 id="genome-hg38">Genome hg38</h2>
<h3 id="sequence">Sequence</h3>
<h4 id="download-from-gencode-v27">download from gencode v27</h4>
<ul>
<li>[ ]located in folder: </li>
</ul>
<h4 id="located-in-folder-sequence">located in folder: sequence/</h4>
<pre><code>mkdir sequence
cd sequence
wget wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/GRCh38.p10.genome.fa.gz
</code></pre>

<h4 id="decompression">Decompression</h4>
<pre><code>gzip -d GRCh38.p10.genome.fa.gz 
</code></pre>

<h4 id="parse-genome">parse genome</h4>
<pre><code>samtools faidx GRCh38.p10.genome.fa
cut -f1,2 GRCh38.p10.genome.fa.fai &gt; hg38.chrom.sizes
</code></pre>

<h3 id="index">Index</h3>
<h4 id="build-by-bowtie2-and-star">build by bowtie2 and STAR</h4>
<ul>
<li>[ ]located in folder: </li>
</ul>
<h4 id="located-in-folder-index">located in folder: index/</h4>
<ul>
<li>using bowtie2</li>
</ul>
<pre><code>mkdir bowtie2_hg38_index
cd bowtie2_hg38_index
bowtie2-build ../sequence/GRCh38.p10.genome.fa GRCh38.p10
</code></pre>

<ul>
<li>using STAR</li>
</ul>
<pre><code>mkdir ../STAR_hg38_index
cd ../STAR_hg38_index
STAR --runMode genomeGenerate --runThreadN 15 --genomeDir . --genomeFastaFiles ../sequence/GRCh38.p10.genome.fa
</code></pre>

<h3 id="annotation-files-for-genome-hg38">Annotation files for genome hg38</h3>
<p>provider: GENCODE and other projects</p>
<p>format: gtf and gff</p>
<h4 id="statistics">statistics</h4>
<table>
<thead>
<tr>
<th align="left"><strong>RNA_type</strong></th>
<th align="left"><strong>gene_num</strong></th>
<th align="left"><strong>transcrips_num</strong></th>
<th align="left"><strong>source</strong></th>
<th align="left"><strong>file</strong></th>
<th align="left"><strong>Date</strong></th>
<th align="left"><strong>Download/Processed</strong></th>
<th align="left"><strong>Note</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">allGenes</td>
<td align="left">58,288</td>
<td align="left">200,401</td>
<td align="left">Gencode27</td>
<td align="left">gencode.v27.annotation.gtf / gencode.v27.annotation.gff</td>
<td align="left">2018.5.4</td>
<td align="left"><a href="#download-gencode-v27-annotations">D/D</a></td>
<td align="left"></td>
</tr>
<tr>
<td align="left">rRNA</td>
<td align="left">544</td>
<td align="left">544</td>
<td align="left">Gencode27</td>
<td align="left">rRNA.gencode27.gtf / rRNA.gencode27.gff</td>
<td align="left">2018.5.4</td>
<td align="left"><a href="#parse-annotations">P/P</a></td>
<td align="left"></td>
</tr>
<tr>
<td align="left">miRNA</td>
<td align="left">1,881</td>
<td align="left">1,881</td>
<td align="left">Gencode27</td>
<td align="left">miRNA.gencode27.gtf / miRNA.gencode27.gff</td>
<td align="left">2018.5.4</td>
<td align="left"><a href="#parse-annotations">P/P</a></td>
<td align="left"></td>
</tr>
<tr>
<td align="left">piRNA</td>
<td align="left">812,347</td>
<td align="left">812,347</td>
<td align="left">piRBase</td>
<td align="left">piRNA.piRBase.hg38.gtf / piRNA.piRBase.hg38.gff</td>
<td align="left">2018.5.4</td>
<td align="left"><a href="#download-piRNA-from-piRBase">D/D</a></td>
<td align="left"></td>
</tr>
<tr>
<td align="left">snoRNA</td>
<td align="left">943</td>
<td align="left">955</td>
<td align="left">Gencode27(misc_RNA)</td>
<td align="left">snoRNA.gencode27.gtf / snoRNA.gencode27.gtf</td>
<td align="left">2018.5.4</td>
<td align="left"><a href="#parse-annotations">P/P</a></td>
<td align="left"></td>
</tr>
<tr>
<td align="left">snRNA</td>
<td align="left">1,900</td>
<td align="left">1,900</td>
<td align="left">Gencode27</td>
<td align="left">snRNA.gencode27.gtf / snRNA.gencode27.gtf</td>
<td align="left">2018.5.4</td>
<td align="left"><a href="#parse-annotations">P/P</a></td>
<td align="left"></td>
</tr>
<tr>
<td align="left">srpRNA</td>
<td align="left">680</td>
<td align="left">680</td>
<td align="left">Gencode27(misc_RNA)</td>
<td align="left">srpRNA.gencode27.gtf / srpRNA.gencode27.gff</td>
<td align="left">2018.5.4</td>
<td align="left"><a href="#parse-annotations">P/P</a></td>
<td align="left"></td>
</tr>
<tr>
<td align="left">tRNA</td>
<td align="left">649</td>
<td align="left">649</td>
<td align="left">Gencode27(predicted tRNA)</td>
<td align="left">tRNA.gencode27.gtf / tRNA.gencode27.gff</td>
<td align="left">2018.5.4</td>
<td align="left"><a href="#download-gencode-v27-annotations">D/D</a></td>
<td align="left"></td>
</tr>
<tr>
<td align="left">lncRNA</td>
<td align="left">15,778</td>
<td align="left">27,908</td>
<td align="left">Gencode27(lncRNA)</td>
<td align="left">lncRNA.gencode27.gtf / lncRNA.gencode27.gff</td>
<td align="left">2018.5.4</td>
<td align="left"><a href="#download-gencode-v27-annotations">D/D</a></td>
<td align="left"></td>
</tr>
<tr>
<td align="left">lncRNA</td>
<td align="left">96,308</td>
<td align="left">172,216</td>
<td align="left">NONCODEv5</td>
<td align="left">lncRNA.NONCODEv5.hg38.gtf / lncRNA.NONCODEv5.hg38.gff</td>
<td align="left">2018.5.4</td>
<td align="left"><a href="#parse-and-convert">P/P</a></td>
<td align="left"></td>
</tr>
<tr>
<td align="left">lncRNA</td>
<td align="left">63,427</td>
<td align="left">174,657</td>
<td align="left">mitranscritome</td>
<td align="left">lncRNA.mitranscriptome.v2.hg38.gtf / lncRNA.mitranscriptome.v2.hg38.gff</td>
<td align="left">2018.5.4</td>
<td align="left"><a href="#parse-and-convert">P/P</a></td>
<td align="left"></td>
</tr>
<tr>
<td align="left">tucp</td>
<td align="left">3,711</td>
<td align="left">11,244</td>
<td align="left">mitranscritome</td>
<td align="left">tucp.mitranscriptome.v2.hg38.gtf / tucp.mitranscriptome.v2.hg38.gff</td>
<td align="left">2018.5.4</td>
<td align="left"><a href="#parse-and-convert">P/P</a></td>
<td align="left"></td>
</tr>
<tr>
<td align="left">lncRNA</td>
<td align="left">131,683</td>
<td align="left">342,295</td>
<td align="left">Gencode27+NONCODEv5+ MiTranscriptome+NC2017</td>
<td align="left">merged_lncRNA.combined.gtf / merged_lncRNA.combined.gff</td>
<td align="left">2018.5.4</td>
<td align="left"><a href="#parse-and-convert">P/P</a></td>
<td align="left"></td>
</tr>
<tr>
<td align="left">mRNA</td>
<td align="left">19,836</td>
<td align="left">80,930</td>
<td align="left">Gencode27(protein_coding)</td>
<td align="left">mRNA.gencode27.gtf / mRNA.gencode27.gff</td>
<td align="left">2018.5.4</td>
<td align="left"><a href="#parse-annotations">P/P</a></td>
<td align="left"></td>
</tr>
</tbody>
</table>
<h4 id="pre-process-annotaion">pre-process annotaion</h4>
<h4 id="download-gencode-v27-annotations">download gencode v27 annotations</h4>
<pre><code>wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/gencode.v27.annotation.gtf.gz
wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/gencode.v27.annotation.gff3.gz
wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/gencode.v27.long_noncoding_RNAs.gtf.gz
wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/gencode.v27.long_noncoding_RNAs.gff3.gz
wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/gencode.v27.tRNAs.gtf.gz
wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/gencode.v27.tRNAs.gff3.gz
</code></pre>

<h4 id="decompression_1">Decompression</h4>
<pre><code>gzip -d *
</code></pre>

<h4 id="parse-annotations">parse annotations</h4>
<pre><code>gtf=gencode.v27.annotation.gtf
for i in rRNA snRNA snoRNA srpRNA miRNA vaultRNA Y_RNA; do
grep &quot;gene_type \&quot;$i\&quot;&quot; $gtf &gt; $i.gencode27.gtf
gffread -E $i.gencode27.gtf -o- &gt; $i.gencode27.gff
done;

grep &quot;gene_type \&quot;protein_coding\&quot;&quot; $gtf | grep -v Selenocysteine &gt; mRNA.gencode27.gtf
grep &quot;RN7SL&quot; $gtf &gt; srpRNA.gencode27.gtf
grep 'Y_RNA' $gtf &gt; Y_RNA.gencode27.gtf 

for i in mRNA srpRNA Y_RNA; do
gffread -E $i.gencode27.gtf -o- &gt; $i.gencode27.gff
done;

mv gencode.v27.tRNAs.gtf tRNA.gencode27.gtf
mv gencode.v27.tRNAs.gff3 tRNA.gencode27.gff

mv gencode.v27.long_noncoding_RNAs.gtf lncRNA.gencode27.gtf
mv gencode.v27.long_noncoding_RNAs.gff3 lncRNA.gencode27.gff

mv gencode.v27.annotation.gff3 gencode.v27.annotation.gff
</code></pre>

<h4 id="download-pirna-from-pirbase">download piRNA from piRBase</h4>
<h4 id="downlaod-lncrna-from-noncodehttpwwwnoncodeorg-mitranscriptomehttpmitranscriptomeorg-and-yang-yangs-nc-paper">downlaod lncRNA from NONCODE(http://www.noncode.org/), mitranscriptome(http://mitranscriptome.org/), and Yang Yang's NC paper</h4>
<pre><code>wget http://mitranscriptome.org/download/mitranscriptome.gtf.tar.gz
wget http://www.noncode.org/datadownload/NONCODEv5_human_hg38_lncRNA.gtf.gz
wget https://media.nature.com/original/nature-assets/ncomms/2017/170213/ncomms14421/extref/ncomms14421-s3.txt
</code></pre>

<h4 id="decompression_2">Decompression</h4>
<pre><code>tar zxvf mitranscriptome.gtf.tar.gz
gzip -d mitranscriptome.v2.gtf.gz
</code></pre>

<h4 id="parse-and-convert">parse and convert</h4>
<pre><code>wget http://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/hg19ToHg38.over.chain.gz
gzip -d hg19ToHg38.over.chain.gz 
cd mitranscriptome.gtf
gzip -d mitranscriptome.v2.gtf.gz
gtfToGenePred -genePredExt mitranscriptome.v2.gtf mitranscriptome.v2.gp
liftOver -genePred mitranscriptome.v2.gp ../hg19ToHg38.over.chain mitranscriptome.v2.hg38.gp unmapped.gtf
genePredToGtf -utr -source=mitranscriptome file mitranscriptome.v2.hg38.gp mitranscriptome.v2.hg38.gtf
gffread -E mitranscriptome.v2.hg38.gtf -o- &gt; mitranscriptome.v2.hg38.gff
mv ../mitranscriptome.gtf ../mitranscriptome
cd ..

# filter out lncRNA
cd mitranscriptome.gtf
zcat mitranscriptome.v2.gtf.gz | grep -e 'tcat \&quot;lncrna\&quot;' &gt; lncRNA.mitranscriptome.v2.gtf
# liftOver hg19 to hg38
gtfToGenePred -genePredExt lncRNA.mitranscriptome.v2.gtf lncRNA.mitranscriptome.v2.gp
liftOver -genePred lncRNA.mitranscriptome.v2.gp ../hg19ToHg38.over.chain lncRNA.mitranscriptome.v2.hg38.gp unmapped.gtf
genePredToGtf -utr -source=mitranscriptome file lncRNA.mitranscriptome.v2.hg38.gp lncRNA.mitranscriptome.v2.hg38.gtf
gffread -E lncRNA.mitranscriptome.v2.hg38.gtf -o- &gt; lncRNA.mitranscriptome.v2.hg38.gff

# filter out tucp
zcat mitranscriptome.v2.gtf.gz | grep -e 'tcat \&quot;tucp\&quot;' | awk '$7!=&quot;.&quot;' &gt; tucp.mitranscriptome.v2.gtf
# liftOver hg19 to hg38
gtfToGenePred -genePredExt tucp.mitranscriptome.v2.gtf tucp.mitranscriptome.v2.gp
liftOver -genePred tucp.mitranscriptome.v2.gp ../hg19ToHg38.over.chain tucp.mitranscriptome.v2.hg38.gp unmapped.gtf
genePredToGtf -utr -source=mitranscriptome file tucp.mitranscriptome.v2.hg38.gp tucp.mitranscriptome.v2.hg38.gtf
gffread -E tucp.mitranscriptome.v2.hg38.gtf -o- &gt; tucp.mitranscriptome.v2.hg38.gff

cd ..
gtfToGenePred -genePredExt ncomms14421-s3.txt lncRNA.lulab_ncomms14421.gp
liftOver -genePred lncRNA.lulab_ncomms14421.gp hg19ToHg38.over.chain lncRNA.lulab_ncomms14421.hg38.gp unmapped.gtf
genePredToGtf -utr -source=lulab_ncomms14421 file lncRNA.lulab_ncomms14421.hg38.gp lncRNA.lulab_ncomms14421.hg38.gtf
gffread -E lncRNA.lulab_ncomms14421.hg38.gtf -o- &gt; lncRNA.lulab_ncomms14421.hg38.gff

cd mitranscriptome.gtf
mv * ..
cd mitranscriptome
mv * ..
rm -rf mitranscriptome mitranscriptome.gtf mitranscriptome.gtf.tar.gz

rm -rf *.gp unmapped.gtf

mv NONCODEv5_human_hg38_lncRNA.gtf lncRNA.NONCODEv5.hg38.gtf
</code></pre>

<h4 id="merge-lncrna-annotations">merge lncRNA annotations</h4>
<pre><code>gffcompare -o merged_lncRNA -s ../sequence/GRCh38.p10.genome.fa lncRNA.NONCODEv5.hg38.gtf  lncRNA.mitranscriptome.v2.hg38.gtf  lncRNA.gencode27.gtf lncRNA.lulab_ncomms14421.hg38.gtf
gffread -E merged_lncRNA.combined.gtf -o- &gt; merged_lncRNA.combined.gff
</code></pre>

<h4 id="mkdir-gtf-gff">mkdir gtf gff</h4>
<pre><code>cd ..
mkdir gtf gff
mv gencode/*.gtf gtf
mv gencode/*.gff gff
</code></pre>

<h4 id="establish-indexes">establish indexes</h4>
<pre><code>if [ &quot;$1&quot; = &quot;0&quot; ]; then
    for i in ${RNAs[@]} ; do
        echo &quot;start $i.gtf:&quot;
        rsem-prepare-reference --gtf $gtf/$i.gtf --bowtie2 $hg38 RNA_index/$i
        echo &quot;$i finished.&quot;
    done
</code></pre>

<h4 id="cut-lncrna-into-bins">cut lncRNA into bins</h4>
<h4 id="convert-combined-gtf-to-combined-bed">convert combined gtf to combined bed</h4>
<pre><code>gffread --bed merged_lncRNA.combined.gtf -o merged_lncRNA.combined.bed
</code></pre>

<h4 id="grep-the-longest-transciprt-for-each-lncrna-gene-and-calculate-its-length">grep the longest transciprt for each lncRNA gene and calculate its length</h4>
<pre><code>awk 'NR==FNR {split($10,m,&quot;\&quot;&quot;);split($12,n,&quot;\&quot;&quot;);D[m[2]]=n[2]} \
NR&gt;FNR {split($11,ex,&quot;,&quot;);for (k in ex) len[$4]+=ex[k]; b[D[$4]] = len[$4] &gt; a[D[$4]] ? $4 : b[D[$4]]; a[D[$4]] = len[$4] &gt; a[D[$4]] ? len[$4] : a[D[$4]] } \
END { OFS = &quot;\t&quot;; for(x in a) print x, a[x],b[x]}' merged_lncRNA.combined.gtf merged_lncRNA.combined.bed &gt;merged_lncRNA.combined.longest.tran
</code></pre>

<h4 id="grep-the-exons-of-the-longest-transciprts">grep the exons of the longest transciprts</h4>
<pre><code>awk '{print $3}' merged_lncRNA.combined.longest.tran| grep -Ff - merged_lncRNA.combined.gtf|awk '$3==&quot;exon&quot;' &gt;merged_lncRNA.combined.longest.exon.gtf
</code></pre>

<h4 id="cut-full-length-lncrna-transcripts-into-bins-bin_size_30-step_size_15">cut full length lncRNA transcripts into bins: bin_size_30, step_size_15</h4>
<h4 id="1deal-with-transcirpts-longer-than-30bp">(1)deal with transcirpts longer than 30bp</h4>
<pre><code>awk 'BEGIN{ OFS=&quot;\t&quot; } ($5-$4)&gt;30 {split($10,a,&quot;\&quot;&quot;); \
for(i=0;i&lt;=($5-$4-30)/15;i++) {print $1,$4+i*15, $4+i*15+30,a[2]&quot;__&quot;$4+i*15&quot;__&quot;$4+i*15+30,$6,$7} \
print $1, $4+(i+1)*15,$5,a[2]&quot;__&quot;$4+(i+1)*15&quot;__&quot;$5,$6,$7}' \
merged_lncRNA.combined.longest.exon.gtf |awk '$3&gt;$2' &gt;merged_lncRNA.combined.longest.exon.bin30.bed
</code></pre>

<h4 id="2deal-with-transcirpts-no-longer-than-30bp">(2)deal with transcirpts no longer than 30bp</h4>
<pre><code>awk 'BEGIN{ OFS=&quot;\t&quot; } ($5-$4)&lt;=30 {split($10,a,&quot;\&quot;&quot;);print $1,$4,$5,a[2]&quot;__&quot;$4&quot;__&quot;$5,$6,$7}' \
merged_lncRNA.combined.longest.exon.gtf | awk '$3&gt;$2' &gt; shorterthan30.bed
</code></pre>

<h4 id="3concatenate-two-bed-files-generated-abovely">(3)concatenate two bed files generated abovely</h4>
<pre><code>cat merged_lncRNA.combined.longest.exon.bin30.bed shorterthan30.bed|sort -k1,1 -k2,2n -k3,3 &gt;merged_lncRNA.combined.longest.exon.bin30.all.bed
awk 'BEGIN{FS=OFS=&quot;\t&quot;}{print $1,$2,$3,$4,1,$6}' merged_lncRNA.combined.longest.exon.bin30.all.bed &gt; foo 
bedToGenePred foo /dev/stdout | genePredToGtf -source=merged_lncRNA_bin file /dev/stdin merged_lncRNA.combined.longest.exon.bin30.all.gtf
</code></pre>

<pre><code>for i in `ls ./gtf`;
do j=${i%.*};
echo $j
gffread --bed ./gtf/$j.gtf -o ./binned/$j.bed

awk 'NR==FNR {split($10,m,&quot;\&quot;&quot;);split($12,n,&quot;\&quot;&quot;);D[m[2]]=n[2]} \
NR&gt;FNR {split($11,ex,&quot;,&quot;);for (k in ex) len[$4]+=ex[k]; b[D[$4]] = len[$4] &gt; a[D[$4]] ? $4 : b[D[$4]]; a[D[$4]] = len[$4] &gt; a[D[$4]] ? len[$4] : a[D[$4]] } \
END { OFS = &quot;\t&quot;; for(x in a) print x, a[x],b[x]}' ./gtf/$j.gtf ./binned/$j.bed &gt; ./binned/$j.longest.tran

awk '{print $3}' ./binned/$j.longest.tran | grep -Ff - ./gtf/$j.gtf | awk '$3==&quot;exon&quot;' &gt; ./binned/$j.longest.exon.gtf

awk 'BEGIN{ OFS=&quot;\t&quot; } ($5-$4)&gt;30 {split($10,a,&quot;\&quot;&quot;); \
for(i=0;i&lt;=($5-$4-30)/15;i++) {print $1,$4+i*15, $4+i*15+30,a[2]&quot;__&quot;$4+i*15&quot;__&quot;$4+i*15+30,$6,$7} \
print $1, $4+(i+1)*15,$5,a[2]&quot;__&quot;$4+(i+1)*15&quot;__&quot;$5,$6,$7}' \
./binned/$j.longest.exon.gtf | awk '$3&gt;$2' &gt; ./binned/$j.longest.exon.bin30.bed

awk 'BEGIN{ OFS=&quot;\t&quot; } ($5-$4)&lt;=30 {split($10,a,&quot;\&quot;&quot;);print $1,$4,$5,a[2]&quot;__&quot;$4&quot;__&quot;$5,$6,$7}' \
./binned/$j.longest.exon.gtf | awk '$3&gt;$2' &gt; ./binned/$j.shorterthan30.bed

cat ./binned/$j.longest.exon.bin30.bed ./binned/$j.shorterthan30.bed | sort -k1,1 -k2,2n -k3,3 &gt; ./binned/$j.longest.exon.bin30.all.bed
awk 'BEGIN{FS=OFS=&quot;\t&quot;}{print $1,$2,$3,$4,1,$6}' ./binned/$j.longest.exon.bin30.all.bed &gt; ./binned/$j.foo
bedToGenePred ./binned/$j.foo /dev/stdout | genePredToGtf -source=${j}_bin file /dev/stdin ./binned/$j.longest.exon.bin30.all.gtf

done
</code></pre>
              
            </div>
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>

</body>
</html>
