<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Genome and Annotations - exseek</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Genome and Annotations";
    var mkdocs_page_input_path = "genome_and_annotations.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> exseek</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Overview</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../get_started/">Get Started</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../installation/">Installation</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../genome_and_annotatins.md">Genome and Annotations</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../small_rna_mapping/">Small RNA-seq mapping</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../long_rna_mapping/">Long RNA-seq mapping</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../matrix_processing/">Matrix processing</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../feature_selection.md">Feature Selection</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../cluster_configuration/">Run on a cluster</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../configuration/">Configuration File Reference</a>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Utilities</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../quality_control/">Quality Control</a>
                </li>
                <li class="">
                    
    <a class="" href="../adapter_removal/">Adapter Removal</a>
                </li>
                <li class="">
                    
    <a class="" href="../bigwig/">Generate BigWig Files</a>
                </li>
                <li class="">
                    
    <a class="" href="../igv_browser/">IGV Browser</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../FAQ/">Frequently Asked Questions</a>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">exseek</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>Genome and Annotations</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="genome-and-annotations">Genome and Annotations</h1>
<h2 id="annotation-summary-table">Annotation summary table</h2>
<table>
<thead>
<tr>
<th>Type</th>
<th>Number of genes</th>
<th>Source</th>
</tr>
</thead>
<tbody>
<tr>
<td>miRNA</td>
<td>1917</td>
<td>miRBase hairpin (Version 22)</td>
</tr>
<tr>
<td>piRNA</td>
<td>23431</td>
<td>piRNABank</td>
</tr>
<tr>
<td>lncRNA</td>
<td>15778</td>
<td>GENCODE V27 and mitranscriptome</td>
</tr>
<tr>
<td>rRNA</td>
<td>37</td>
<td>NCBI refSeq 109</td>
</tr>
<tr>
<td>mRNA</td>
<td>19836</td>
<td>GENCODE V27</td>
</tr>
<tr>
<td>snoRNA</td>
<td>943</td>
<td>GENCODE V27</td>
</tr>
<tr>
<td>snRNA</td>
<td>1900</td>
<td>GENCODE V27</td>
</tr>
<tr>
<td>srpRNA</td>
<td>680</td>
<td>GENCODE V27</td>
</tr>
<tr>
<td>tRNA</td>
<td>649</td>
<td>GENCODE V27</td>
</tr>
<tr>
<td>tucpRNA</td>
<td>3734</td>
<td>GENCODE V27</td>
</tr>
<tr>
<td>Y_RNA</td>
<td>756</td>
<td>GENCODE V27</td>
</tr>
<tr>
<td>circRNA</td>
<td>140527</td>
<td>circBase</td>
</tr>
<tr>
<td>repeats</td>
<td>-</td>
<td>UCSC Genome Browser (rmsk)</td>
</tr>
<tr>
<td>promoter</td>
<td>-</td>
<td>ChromHMM tracks from 9 cell lines from UCSC Genome Browser</td>
</tr>
<tr>
<td>enhancer</td>
<td>-</td>
<td>ChromHMM tracks from 9 cell lines from UCSC Genome Browser</td>
</tr>
</tbody>
</table>
<h2 id="genome-and-annotation-files">Genome and annotation files</h2>
<table>
<thead>
<tr>
<th>File</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>fasta/genome.fa</code></td>
<td>genome sequence</td>
</tr>
<tr>
<td><code>fasta/circRNA.fa</code></td>
<td>junction sequence in circBase</td>
</tr>
<tr>
<td><code>fasta/rRNA.fa</code></td>
<td>rRNA sequences in NCBI RefSeq</td>
</tr>
<tr>
<td><code>fasta/miRNA.fa</code></td>
<td>miRNA hairpin (precursor) sequences in miRBase</td>
</tr>
<tr>
<td><code>fasta/piRNA.fa</code></td>
<td>piRNA sequences in piRNABank</td>
</tr>
<tr>
<td><code>fasta/${rna_type}.fa</code></td>
<td>Longest isoform for each gene extracted from GENCODE annotations</td>
</tr>
<tr>
<td><code>gtf_by_biotype/${rna_type}.gtf</code></td>
<td>separate GTF files for each RNA type</td>
</tr>
<tr>
<td><code>gtf/gencode.gtf</code></td>
<td>GENCODE GTF file</td>
</tr>
<tr>
<td><code>gtf/mitranscriptome.gtf</code></td>
<td>Mitranscriptome GTF file</td>
</tr>
<tr>
<td><code>gtf/long_RNA.gtf</code></td>
<td>GTF file of Long RNA (GENCODE + Mitranscriptome - miRNA)</td>
</tr>
<tr>
<td><code>gtf/piRNABank.gtf</code></td>
<td>piRNA GTF file from piRNABank</td>
</tr>
<tr>
<td><code>gtf/gencode_tRNA.gtf</code></td>
<td>GTF file of tRNA from GENCODE</td>
</tr>
<tr>
<td><code>transcript_table/all.txt</code></td>
<td>Table of transcript information (gene_id, transcript_id)</td>
</tr>
<tr>
<td><code>rsem_index/bowtie2/${rna_type}</code></td>
<td>RSEM index files for each RNA type (built using the longest transcripts)</td>
</tr>
<tr>
<td><code>rsem_index/bowtie2/${rna_type}.transcripts.fa</code></td>
<td>Sequence for each RNA type (longest transcripts)</td>
</tr>
<tr>
<td><code>gtf_longest_transcript/${rna_type}.gtf</code></td>
<td>GTF files for the longest isoforms from GENCODE and Mitranscriptome</td>
</tr>
<tr>
<td><code>bed/*.bed</code></td>
<td>Transcript in BED12 format extracted from GTF files in `gtf/*.gtf</td>
</tr>
<tr>
<td><code>index/bowtie2/${rna_type}</code></td>
<td>STAR index for transcripts</td>
</tr>
<tr>
<td><code>index/star/${rna_type}</code></td>
<td>STAR index for transcripts</td>
</tr>
<tr>
<td><code>long_index/star/</code></td>
<td>STAR index including splicing junctions of long RNA</td>
</tr>
</tbody>
</table>
<h2 id="generate-the-genome-and-annotation-files">Generate the genome and annotation files</h2>
<h3 id="create-genome-directory">Create genome directory</h3>
<pre><code class="bash">[ -d &quot;genome/hg38/source&quot; ] || mkdir -p &quot;genome/hg38/source&quot;
</code></pre>

<h3 id="chromosome-id-conversion-table">Chromosome ID conversion table</h3>
<ul>
<li>Column 1: UCSC chromosome ID</li>
<li>Column 2: RefSeq chromosome ID</li>
</ul>
<pre><code class="bash">mysql --user=genome --host=genome-mysql.cse.ucsc.edu -A -N -e &quot;SELECT * FROM ucscToRefSeq;&quot; hg38 | cut -f1,4 &gt; genome/hg38/source/ucscToRefSeq.txt
</code></pre>

<h3 id="download-gene-annotation-ncbi">Download Gene annotation (NCBI)</h3>
<pre><code class="bash"># NCBI Human Release 109
wget -P genome/hg38/source ftp://ftp.ncbi.nlm.nih.gov/genomes/H_sapiens/GFF/ref_GRCh38.p12_top_level.gff3.gz
[ -d genome/hg38/gff3 ] || mkdir -p genome/hg38/gff3
awk 'BEGIN{OFS=&quot;\t&quot;;FS=&quot;\t&quot;} NR==FNR{c[$2]=$1;next} !/^#/{chrom=c[$1]; if(length(chrom) &gt; 0) print c[$1],$2,$3,$4,$5,$6,$7,$8,$9}' \
    genome/hg38/source/ucscToRefSeq.txt &lt;(zcat genome/hg38/source/ref_GRCh38.p12_top_level.gff3.gz) \
    &gt; genome/hg38/gff3/refseq.gff3
gffread --bed -o genome/hg38/bed/ncbi.bed genome/hg38/gff3/refseq.gff3
wget -O genome/hg38/source/refSeq_rna.fa.gz ftp://ftp.ncbi.nlm.nih.gov/genomes/H_sapiens/RNA/rna.fa.gz
# get rRNA sequence IDs
zgrep 'ribosomal RNA$' genome/hg38/source/refSeq_rna.fa.gz \
    | sed 's/&gt;ref|\(NR_[0-9.]\+\)|.*(\([^)]\+\)).*/\1|\2/' &gt; genome/hg38/source/refSeq_rRNA.ids.txt
# get rRNA sequences
zcat genome/hg38/source/refSeq_rna.fa.gz \
    | awk 'FNR==NR{split($0,a,&quot;|&quot;);ids[a[1]]=1;next} 
{if($0 ~ /^&gt;/){split($0,a,&quot;|&quot;);if(ids[a[2]] == 1){keep=1; print &quot;&gt;&quot; a[2];}else{keep=0}} else{if(keep == 1){print}}}' \
genome/hg38/source/refSeq_rRNA.ids.txt - &gt; genome/hg38/fasta/rRNA.fa
samtools faidx genome/hg38/fasta/rRNA.fa
# generate transcript table
{
echo -e 'chrom\tstart\tend\tname\tscore\tstrand\tgene_id\ttranscript_id\tgene_name\ttranscript_name\tgene_type\ttranscript_type\tsource'
awk 'BEGIN{OFS=&quot;\t&quot;;FS=&quot;\t&quot;}FNR==NR{split($0,a,&quot;|&quot;);gene_name[a[1]]=a[2];next}{print $1,0,$2,a[1],0,&quot;+&quot;,$1,$1,gene_name[$1],gene_name[$1],&quot;rRNA&quot;,&quot;rRNA&quot;,&quot;RefSeq&quot;}' \
    genome/hg38/source/refSeq_rRNA.ids.txt genome/hg38/fasta/rRNA.fa.fai
} &gt; genome/hg38/transcript_table/rRNA.txt
# get transcript sizes
cut -f1,2 genome/hg38/fasta/rRNA.fa.fai &gt; genome/hg38/chrom_sizes/rRNA
# build STAR index (small genome)
STAR --runMode genomeGenerate --genomeSAindexNbases 5 --genomeDir genome/hg38/index/star/rRNA/ --genomeFastaFiles genome/hg38/fasta/rRNA.fa
</code></pre>

<h3 id="download-chain-files-for-crossmap">Download chain files for CrossMap</h3>
<pre><code class="bash">wget -O genome/hg38/source/hg18ToHg38.over.chain.gz http://hgdownload.soe.ucsc.edu/goldenPath/hg18/liftOver/hg18ToHg38.over.chain.gz
wget -O genome/hg38/source/NCBI36_to_GRCh38.chain.gz https://sourceforge.net/projects/crossmap/files/Ensembl_chain_files/homo_sapiens%28human%29/NCBI36_to_GRCh38.chain.gz
</code></pre>

<h3 id="genome-assembly-ucsc-hg38">Genome assembly (UCSC hg38)</h3>
<pre><code class="bash">wget -P genome/hg38/source http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz
gzip -d -c genome/hg38/source/hg38.fa.gz &gt; genome/hg38/fasta/genome.fa
samtools faidx genome/hg38/fasta/genome.fa
</code></pre>

<h3 id="encode-annotations">ENCODE annotations</h3>
<pre><code class="bash">wget -P genome/hg38/source ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/gencode.v27.annotation.gtf.gz
#wget -P genome/hg38/source ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/gencode.v27.annotation.gff3.gz
wget -P genome/hg38/source ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/gencode.v27.long_noncoding_RNAs.gtf.gz
#wget -P genome/hg38/source ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/gencode.v27.long_noncoding_RNAs.gff3.gz
wget -P genome/hg38/source ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/gencode.v27.tRNAs.gtf.gz
#wget -P genome/hg38/source ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_27/gencode.v27.tRNAs.gff3.gz
zcat genome/hg38/source/gencode.v27.annotation.gtf.gz &gt; genome/hg38/gtf/gencode.gtf
zcat genome/hg38/source/gencode.v27.long_noncoding_RNAs.gtf.gz &gt; genome/hg38/gtf/gencode_lncRNA.gtf
zcat genome/hg38/source/gencode.v27.tRNAs.gtf.gz \
    | awk 'BEGIN{FS=&quot;\t&quot;;OFS=&quot;\t&quot;}{print $1,$2,&quot;exon&quot;,$4,$5,$6,$7,$8,$9}' &gt; genome/hg38/gtf/gencode_tRNA.gtf
# Chain file for converting hg19 to hg38
wget -P genome/hg38/source http://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/hg19ToHg38.over.chain.gz
</code></pre>

<h3 id="mitranscriptome">Mitranscriptome</h3>
<pre><code class="bash">wget -P genome/hg38/source http://mitranscriptome.org/download/mitranscriptome.gtf.tar.gz
tar -C genome/hg38/source --strip-components=1 -zxf genome/hg38/source/mitranscriptome.gtf.tar.gz mitranscriptome.gtf/mitranscriptome.v2.gtf.gz
# convert from hg19 to hg38
zcat genome/hg38/source/mitranscriptome.v2.gtf.gz \
    | CrossMap.py gff genome/hg38/source/hg19ToHg38.over.chain.gz /dev/stdin genome/hg38/source/mitranscriptome.v2.hg38.gtf
# remove invalid transcripts
bin/preprocess.py fix_gtf -i genome/hg38/source/mitranscriptome.v2.hg38.gtf -o genome/hg38/gtf/mitranscriptome.gtf
</code></pre>

<p>Extract lncRNA and TUCP RNA to separate GTF files:</p>
<pre><code class="bash">grep 'tcat &quot;lncrna&quot;' genome/hg38/gtf/mitranscriptome.gtf &gt; genome/hg38/gtf/mitranscriptome_lncRNA.gtf
# add exon feature
grep 'tcat &quot;tucp&quot;' genome/hg38/gtf/mitranscriptome.gtf \
    | awk 'BEGIN{OFS=&quot;\t&quot;;FS=&quot;\t&quot;}{print;print $1,$2,&quot;exon&quot;,$4,$5,$6,$7,$8,$9}' &gt; genome/hg38/gtf/mitranscriptome_tucp.gtf
cp genome/hg38/gtf/mitranscriptome_tucp.gtf genome/hg38/gtf_by_biotype/tucpRNA.gtf
</code></pre>

<h3 id="noncode">NONCODE</h3>
<pre><code class="bash">wget -P genome/hg38/source http://www.noncode.org/datadownload/NONCODEv5_human_hg38_lncRNA.gtf.gz
zcat genome/hg38/source/NONCODEv5_human_hg38_lncRNA.gtf.gz \
    | awk 'BEGIN{FS=&quot;\t&quot;;OFS=&quot;\t&quot;}$7 != &quot;.&quot; {print $1,&quot;NONCODE&quot;,$3,$4,$5,$6,$7,$8,$9}' &gt; genome/hg38/gtf/noncode.gtf
</code></pre>

<h3 id="lncrnas-identified-in-hcc-nature-communications-2017">lncRNAs identified in HCC (Nature communications 2017)</h3>
<pre><code class="bash">wget -P genome/hg38/source https://media.nature.com/original/nature-assets/ncomms/2017/170213/ncomms14421/extref/ncomms14421-s3.txt
awk 'BEGIN{FS=&quot;\t&quot;;OFS=&quot;\t&quot;}{print $1,&quot;ncomms2017&quot;,$3,$4,$5,$6,$7,$8,$9}' genome/hg38/source/ncomms14421-s3.txt &gt; genome/hg38/source/ncomms2017.gtf
CrossMap.py gff genome/hg38/source/hg19ToHg38.over.chain.gz genome/hg38/source/ncomms2017.gtf genome/hg38/source/ncomms2017.hg38.gtf
ln genome/hg38/source/ncomms2017.hg38.gtf genome/hg38/gtf/ncomms2017.gtf
</code></pre>

<h3 id="merge-lncrna-gencode-and-mitranscriptome">Merge lncRNA (GENCODE and Mitranscriptome)</h3>
<pre><code class="bash">cat genome/hg38/gtf/gencode_lncRNA.gtf \
    genome/hg38/gtf/mitranscriptome_lncRNA.gtf \
    &gt; genome/hg38/gtf/merged_lncRNA.gtf
cp genome/hg38/gtf/merged_lncRNA.gtf genome/hg38/gtf_by_biotype/lncRNA.gtf
</code></pre>

<h3 id="pirbase-v10">piRBase (v1.0)</h3>
<pre><code class="bash">wget -O genome/hg38/source/piRBase-hsa-v1.0.bed.gz http://www.regulatoryrna.org/database/piRNA/download/archive/v1.0/bed/piR_hg19_v1.0.bed.gz
zcat genome/hg38/source/piRBase-hsa-v1.0.bed.gz \
    | CrossMap.py bed genome/hg38/source/hg19ToHg38.over.chain.gz /dev/stdin genome/hg38/source/piRBase-hsa-v1.0.hg38.bed
bedToGenePred genome/hg38/source/piRBase-hsa-v1.0.hg38.bed genome/hg38/source/piRBase-hsa-v1.0.hg38.genePred
genePredToGtf -source=piRBase file genome/hg38/source/piRBase-hsa-v1.0.hg38.genePred genome/hg38/source/piRBase-hsa-v1.0.hg38.gtf
ln genome/hg38/source/piRBase-hsa-v1.0.hg38.gtf genome/hg38/gtf/piRBase.gtf
</code></pre>

<h3 id="pirbase-v20">piRBase (v2.0)</h3>
<pre><code class="bash">wget -O genome/hg38/source/piRBase-hsa-v2.0.bed.gz http://www.regulatoryrna.org/database/piRNA/download/archive/v2.0/bed/hsa.bed.gz
zcat genome/hg38/source/piRBase-hsa-v2.0.bed.gz | bedtools sort &gt; source/piRBase-hsa-v2.0.bed
bedToGenePred source/piRBase-hsa-v2.0.bed source/piRBase-hsa-v2.0.genePred
genePredToGtf -source=piRBase file source/piRBase-hsa-v2.0.genePred source/piRBase-hsa-v2.0.gtf
</code></pre>

<h3 id="long-rna-gencode-mitranscriptome-mirna">Long RNA (GENCODE + Mitranscriptome - miRNA)</h3>
<pre><code class="bash"># Merge GTF files
cat genome/hg38/gtf/gencode.gtf \
    genome/hg38/gtf/mitranscriptome_lncRNA.gtf \
    genome/hg38/gtf/mitranscriptome_tucp.gtf \
    | grep -v 'gene_type &quot;miRNA' \
    &gt; genome/hg38/gtf/long_RNA.gtf
# Get gene lengths
tools/GTFtools_0.6.5/gtftools.py -c 1-22,X,Y,M -l genome/hg38/gene_length/long_RNA genome/hg38/gtf/long_RNA.gtf
# GTF to BED12 format
gffread --bed -o genome/hg38/bed/long_RNA.bed genome/hg38/gtf/long_RNA.gtf
</code></pre>

<p><strong>gene_length/long_RNA</strong></p>
<ul>
<li>Tab-deliminated text file</li>
<li>First row: header</li>
<li>Column 1 (gene): gene_id</li>
<li>Column 2 (mean): mean length of isoforms</li>
<li>Column 3 (median): median length of isoforms</li>
<li>Column 4 (longest_isoform): length of the longest isoform</li>
<li>Column 5 (merged): merged length of isoforms</li>
</ul>
<h3 id="pirnabank-ncbi36">piRNABank (NCBI36)</h3>
<pre><code class="bash">wget -O genome/hg38/source/ http://pirnabank.ibab.ac.in/downloads/all/human_all.zip
unzip genome/hg38/source/human_all.zip -d genome/hg38/source/
mv genome/hg38/source/human_pir.txt genome/hg38/source/piRNABank.human.txt
# Extract genomic coordinates from piRNABank
awk 'BEGIN{OFS=&quot;\t&quot;}
    /^&gt;/{na=split(substr($0,2),a,&quot;|&quot;);split(a[na],b,&quot;:&quot;); 
    if(b[5]==&quot;Plus&quot;){s=&quot;+&quot;} else{s=&quot;-&quot;}
    if(a[1]!=name){print &quot;chr&quot; b[2],b[3]-1,b[4],a[1],0,s}
    name=a[1]}' genome/hg38/source/piRNABank.human.txt \
    | bedtools sort &gt; genome/hg38/source/piRNABank.human.bed
awk 'BEGIN{OFS=&quot;\t&quot;}
    {if($0 ~ /^&gt;/) {split(substr($0,2),a,&quot;|&quot;); 
        if((a[1] != name)&amp;&amp;(length(seq) &gt; 0)){print &quot;&gt;&quot; name;gsub(/U/,&quot;T&quot;,seq);print seq} name=a[1]}
    else{seq=$0}}' genome/hg38/source/piRNABank.human.txt &gt; genome/hg38/source/piRNABank.human.fa
bedToGenePred genome/hg38/source/piRNABank.human.bed genome/hg38/source/piRNABank.human.genePred
genePredToGtf -source=piRNABank file genome/hg38/source/piRNABank.human.genePred stdout \
    | awk '$3==&quot;exon&quot;' &gt; genome/hg38/source/piRNABank.human.gtf

CrossMap.py gff genome/hg38/source/hg18ToHg38.over.chain.gz genome/hg38/source/piRNABank.human.gtf \
    genome/hg38/source/piRNABank.human.hg38.gtf
cp genome/hg38/source/piRNABank.human.hg38.gtf genome/hg38/gtf/piRNABank.gtf
cp genome/hg38/gtf/piRNABank.gtf genome/hg38/gtf_by_biotype/piRNA.gtf
gffread --bed -o genome/hg38/source/piRNABank.human.hg38.bed genome/hg38/source/piRNABank.human.hg38.gtf
bedtools getfasta -s -name -fi genome/hg38/fasta/genome.fa -bed genome/hg38/source/piRNABank.human.hg38.bed -split \
    &gt; genome/hg38/source/piRNABank.human.hg38.fa
</code></pre>

<h3 id="mirbase-version-22">miRBase (Version 22)</h3>
<pre><code class="bash">wget -O genome/hg38/source/miRBase.hsa.gff3 ftp://mirbase.org/pub/mirbase/CURRENT/genomes/hsa.gff3
wget -O genome/hg38/source/miRBase.hairpin.fa.gz ftp://mirbase.org/pub/mirbase/CURRENT/hairpin.fa.gz
wget -O genome/hg38/source/miRBase.mature.fa.gz ftp://mirbase.org/pub/mirbase/CURRENT/mature.fa.gz

cp genome/hg38/source/miRBase.hsa.gff3 genome/hg38/gtf/miRBase.gff3
# extract human pre-miRNA
zcat genome/hg38/source/miRBase.hairpin.fa.gz \
    | awk '/^&gt;/{if($0 ~ /&gt;hsa-/) {keep=1; print $1} else{keep=0}; next}{if(keep==1){gsub(/U/, &quot;T&quot;);print}}' \
    &gt; genome/hg38/fasta/miRNA.fa
samtools faidx genome/hg38/fasta/miRNA.fa
# extract human mature miRNA
zcat genome/hg38/source/miRBase.mature.fa.gz \
    | awk '/^&gt;/{if($0 ~ /&gt;hsa-/) {keep=1; print $1} else{keep=0}; next}{if(keep==1){gsub(/U/, &quot;T&quot;);print}}' \
    &gt; genome/hg38/fasta/mature_miRNA.fa
samtools faidx genome/hg38/fasta/mature_miRNA.fa
# generate transcript table (mature miRNA)
{
echo -e 'chrom\tstart\tend\tname\tscore\tstrand\tgene_id\ttranscript_id\tgene_name\ttranscript_name\tgene_type\ttranscript_type\tsource'
awk 'BEGIN{OFS=&quot;\t&quot;;FS=&quot;\t&quot;}{print $1,0,$2,$1,0,&quot;+&quot;,$1,$1,$1,$1,&quot;miRNA&quot;,&quot;miRNA&quot;,&quot;miRBase&quot;}' \
    genome/hg38/fasta/miRNA.fa.fai genome/hg38/fasta/mature_miRNA.fa.fai 
} &gt; genome/hg38/transcript_table/miRNA.txt
# get transcript sizes
cut -f1,2 genome/hg38/fasta/miRNA.fa.fai genome/hg38/fasta/mature_miRNA.fa.fai &gt; genome/hg38/chrom_sizes/miRNA
# gff3 to genePred
awk 'BEGIN{OFS=&quot;\t&quot;;FS=&quot;\t&quot;;d[&quot;miRNA&quot;]=&quot;transcript&quot;;d[&quot;miRNA_primary_transcript&quot;]=&quot;primary_transcript&quot;}/^#/{print}!/^#/{$3=d[$3];print $1,$2,$3,$4,$5,$6,$7,$8,$9}' \
    genome/hg38/gff3/miRBase.gff3 &gt; genome/hg38/source/miRBase.fixed.gff3
gff3ToGenePred -useName genome/hg38/source/miRBase.fixed.gff3 genome/hg38/genePred/miRBase.genePred
</code></pre>

<h3 id="intron">Intron</h3>
<pre><code class="bash">bin/preprocess.py extract_gene -i genome/hg38/gtf/long_RNA.gtf | bedtools sort &gt; genome/hg38/bed/long_RNA.gene.bed
awk 'BEGIN{OFS=&quot;\t&quot;;FS=&quot;\t&quot;} !/^#/{match($9,/gene_id &quot;([^&quot;]+)&quot;/,a);print $1,$4-1,$5,a[1],0,$7}' genome/hg38/gtf/long_RNA.gtf \
    | bedtools sort &gt; genome/hg38/bed/long_RNA.exon.bed
bedtools subtract -sorted -s -a genome/hg38/bed/long_RNA.gene.bed -b genome/hg38/bed/long_RNA.exon.bed \
    | bedtools sort &gt; genome/hg38/bed/long_RNA.intron.bed
</code></pre>

<h3 id="promoterenhancer-from-chromhmm-hg19">Promoter/enhancer from ChromHMM (hg19)</h3>
<pre><code class="bash">wget -P genome/hg38/source http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHmm/wgEncodeBroadHmmGm12878HMM.bed.gz
wget -P genome/hg38/source http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHmm/wgEncodeBroadHmmH1hescHMM.bed.gz
wget -P genome/hg38/source http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHmm/wgEncodeBroadHmmHepg2HMM.bed.gz
wget -P genome/hg38/source http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHmm/wgEncodeBroadHmmHmecHMM.bed.gz
wget -P genome/hg38/source http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHmm/wgEncodeBroadHmmHsmmHMM.bed.gz
wget -P genome/hg38/source http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHmm/wgEncodeBroadHmmHuvecHMM.bed.gz
wget -P genome/hg38/source http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHmm/wgEncodeBroadHmmK562HMM.bed.gz
wget -P genome/hg38/source http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHmm/wgEncodeBroadHmmNhekHMM.bed.gz
wget -P genome/hg38/source http://hgdownload.soe.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeBroadHmm/wgEncodeBroadHmmNhlfHMM.bed.gz

# hg19 =&gt; hg38
tracks=&quot;wgEncodeBroadHmmGm12878HMM wgEncodeBroadHmmH1hescHMM wgEncodeBroadHmmHepg2HMM
    wgEncodeBroadHmmHmecHMM wgEncodeBroadHmmHsmmHMM wgEncodeBroadHmmHuvecHMM
    wgEncodeBroadHmmK562HMM wgEncodeBroadHmmNhekHMM wgEncodeBroadHmmNhlfHMM&quot;
for track in $tracks;do
    CrossMap.py bed genome/hg38/source/hg18ToHg38.over.chain.gz &lt;(zcat genome/hg38/source/${track}.bed.gz) genome/hg38/source/${track}.hg38.bed
    awk 'BEGIN{OFS=&quot;\t&quot;;FS=&quot;\t&quot;}($4==&quot;1_Active_Promoter&quot;)||($4==&quot;2_Weak_Promoter&quot;)||($4==&quot;3_Poised_Promoter&quot;){print $1,$2,$3,$4,$5,$6}' \
        genome/hg38/source/${track}.hg38.bed | bedtools sort &gt; genome/hg38/bed/promoter.${track}.bed
    awk 'BEGIN{OFS=&quot;\t&quot;;FS=&quot;\t&quot;}($4==&quot;4_Strong_Enhancer&quot;)||($4==&quot;5_Strong_Enhancer&quot;)||($4==&quot;6_Weak_Enhancer&quot;)||($4==&quot;7_Weak_Enhancer&quot;){print $1,$2,$3,$4,$5,$6}' \
        genome/hg38/source/${track}.hg38.bed | bedtools sort &gt; genome/hg38/bed/enhancer.${track}.bed
done
# merge promoters and enhancers from 9 cell lines
cat $(for track in $tracks;do echo genome/hg38/bed/promoter.${track}.bed;done) \
    | bedtools sort | bedtools merge -d 1 \
    | awk 'BEGIN{OFS=&quot;\t&quot;;FS=&quot;\t&quot;}{print $1,$2,$3,&quot;promoter&quot;,0,&quot;.&quot;}' &gt; genome/hg38/bed/promoter.merged.bed
cat $(for track in $tracks;do echo genome/hg38/bed/enhancer.${track}.bed;done) \
    | bedtools sort | bedtools merge -d 1 \
    | awk 'BEGIN{OFS=&quot;\t&quot;;FS=&quot;\t&quot;}{print $1,$2,$3,&quot;enhancer&quot;,0,&quot;.&quot;}' &gt; genome/hg38/bed/enhancer.merged.bed
</code></pre>

<h3 id="repeats">Repeats</h3>
<p>UCSC GenomeBrowser -&gt; Tools -&gt; Table Browser</p>
<ul>
<li>assembly: GRCh38/hg38</li>
<li>group: repeats</li>
<li>track: RepeatMasker</li>
<li>table: rmsk</li>
</ul>
<p>Dowload to: genome/hg38/source/rmsk.bed.gz</p>
<pre><code class="bash">gunzip -c genome/hg38/source/rmsk.bed.gz | bedtools sort &gt; genome/hg38/bed/rmsk.bed
</code></pre>

<h3 id="circrna-database-circbase">circRNA database (circBase)</h3>
<pre><code class="bash">wget -O genome/hg38/source/circbase.hg19.fa.gz http://www.circbase.org/download/human_hg19_circRNAs_putative_spliced_sequence.fa.gz
zcat genome/hg38/source/circbase.hg19.fa.gz | bin/preprocess.py extract_circrna_junction -s 150 -o genome/hg38/fasta/circRNA.fa
samtools faidx genome/hg38/fasta/circRNA.fa
STAR --runMode genomeGenerate --genomeSAindexNbases 10 --genomeChrBinNbits 7 --genomeDir genome/hg38/index/star/circRNA/ --genomeFastaFiles genome/hg38/fasta/circRNA.fa
</code></pre>

<h3 id="merge-transcript-table">Merge transcript table</h3>
<pre><code class="bash">{
echo -e 'chrom\tstart\tend\tname\tscore\tstrand\tgene_id\ttranscript_id\tgene_name\ttranscript_name\tgene_type\ttranscript_type\tsource'
for rna_type in rRNA lncRNA miRNA mRNA piRNA snoRNA snRNA srpRNA tRNA tucpRNA Y_RNA;do
    sed '1 d' genome/hg38/transcript_table/${rna_type}.txt
done
} &gt; genome/hg38/transcript_table/all.txt
</code></pre>
              
            </div>
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>

</body>
</html>
