<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Configuration File Reference - exseek</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Configuration File Reference";
    var mkdocs_page_input_path = "configuration.md";
    var mkdocs_page_url = "/configuration/";
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> exseek</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Overview</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../get_started/">Get Started</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../installation/">Installation</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../genome_and_annotations/">Genome and Annotations</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../small_rna_mapping/">Small RNA-seq mapping</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../long_rna_mapping/">Long RNA-seq mapping</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../matrix_processing/">Matrix processing</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../feature_selection/">Feature Selection</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../cluster_configuration/">Run on a cluster</a>
	    </li>
          
            <li class="toctree-l1 current">
		
    <a class="current" href="./">Configuration File Reference</a>
    <ul class="subnav">
            
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Utilities</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../quality_control/">Quality Control</a>
                </li>
                <li class="">
                    
    <a class="" href="../adapter_removal/">Adapter Removal</a>
                </li>
                <li class="">
                    
    <a class="" href="../bigwig/">Generate BigWig Files</a>
                </li>
                <li class="">
                    
    <a class="" href="../igv_browser/">IGV Browser</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../FAQ/">Frequently Asked Questions</a>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">exseek</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>Configuration File Reference</li>
    <li class="wy-breadcrumbs-aside">
      
        <a href="https://github.com/lulab/exSeek/edit/master/docs/configuration.md"
          class="icon icon-github"> Edit on GitHub</a>
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <pre><code class="yaml"># RNA types for sequential mapping in small-RNA pipeline
rna_types: [rRNA, lncRNA, miRNA, mRNA, piRNA, snoRNA, 
  snRNA, srpRNA, tRNA, tucpRNA, Y_RNA]
# Adjusted p-value threshold for defining domains
call_domain_pvalue: &quot;05&quot;
# Distribution to use to model read coverage in each bin
distribution: ZeroTruncatedNegativeBinomial
# Size of each bin to compute read coverage
bin_size: 20
# Define recurrent domain as domains called in fraction of samples above this value
cov_threshold: 0.2
# Method to scale features
scale_method: robust
# Classifier for feature selection
classifiers: random_forest
# Number of features to select
n_selects: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 20, 30, 40, 50]
resample_method: bootstrap
# Feature selection methods
select_methods: [robust]

# Parameters for classifiers
classifier_params:
  logistic_regression:
    penalty: l2
  random_forest:
    n_estimators: 10
# Parameters for grid search for classifier parameters
grid_search:
  logistic_regression:
    C: [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 1e2, 1e3, 1e4, 1e5]
  linear_svm:
    C: [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 1e2, 1e3, 1e4, 1e5]
  random_forest:
    n_estimators: [25, 50, 75]
    max_depth: [3, 4, 5, 6, 7, 8]
# Splitter for cross-validation
cv_splitter: stratified_shuffle_split
# Number of cross-validation folds for grid search
grid_search_n_splits: 5
# Number of train-test splits for cross-validation
cv_n_splits: 50
# Fraction/number of test samples in cross-validation
cv_test_size: 0.2
# Compute sample weight for imbalanced classes
compute_sample_weight: true
# Fold change filter in feature selection (up, down, both)
fold_change_direction: up
# Fold change filter threshold
fold_change_threshold: 1
# Fraction of features to eliminate in each RFE step
rfe_step: 0.1
# Number of cross-validation splits in RFE
rfe_n_splits: 10
# Number of cross-validation splits in robust feature selection
robust_feature_selection_n_splits: 10
# Splitter for robust feature selection
robust_feature_selection_splitter: stratified_shuffle_split

# Number of random train-test splits for cross-validation
cross_validation_splits: 50
# Type of counts for feature selection
#   domains_combined: combine miRNA/piRNA with long RNA domains
#   transcript: transcript-level features
#   featurecounts: gene-level features counted using featureCounts
count_method: domains_combined
# Define low expression value as read counts below this value
filtercount: 10
# Keep features with high expression in fraction of samples above this value
filtersample: 0.2
# Imputation methods to try (set to &quot;null&quot; to skip imputation)
imputation_methods: [&quot;scimpute_count&quot;, &quot;viper_count&quot;, &quot;null&quot;]
# Read depth normalization methods to try
normalization_methods: [&quot;SCnorm&quot;, &quot;TMM&quot;, &quot;RLE&quot;, &quot;CPM&quot;, &quot;CPM_top&quot;, &quot;CPM_rm&quot;]
# Batch effect removal methods to try (set &quot;null&quot; to skip batch effect removal)
batch_removal_methods: [&quot;null&quot;, &quot;Combat&quot;, &quot;RUV&quot;]
# Column index of batch effect in batch_info.txt to considier for Combat
batch_indices: []

# Root directory
root_dir: &quot;.&quot;
# Directory for sequences and annotations
genome_dir: &quot;genome/hg38&quot;
# Temporary directory (e.g. samtools sort, sort)
temp_dir: &quot;tmp&quot;
# Directory for third-party tools
tools_dir: &quot;tools&quot;
# Directory for exSeek scripts
bin_dir: &quot;bin&quot;

# Number of threads for uncompression and compression
threads_compress: 1
# Default number of threads to use
threads: 1
# alignment software to use (valie choices: bowtie, star)
aligner: bowtie2
# Remove 3'-end adaptor sequence from single-end reads
adaptor: &quot;&quot;
# Remove 5'-end adaptor sequence from single-end reads
adaptor_5p: &quot;&quot;
# Remove 3'-end adaptor sequence from the first read in a pair
adaptor1: &quot;&quot;
# Remove 3'-end adaptor sequence from the second read in a pair
adaptor2: &quot;&quot;
# Remove 5'-end adaptor sequence from the first read in a pair
adaptor1_5p: &quot;&quot;
# Remove 5'-end adaptor sequence from the second in a pair
adaptor2_5p: &quot;&quot;
# Exact number of bases to trim from 5'-end
trim_5p: 0
# Exact number of bases to trim from 3'-end
trim_3p: 0
# Discard reads of length below this value
min_read_length: 16
# Maximum read length
max_read_length: 100
# Trim bases with quality below this value from 3'-end
min_base_quality: 30
# Trim bases with quality below this value from 5'-end
min_base_quality_5p: 30
# Trim bases with quality below this value from 3'-end
min_base_quality_3p: 30
# Quality encoding in FASTQ files
quality_base: 33
# Strandness (valid choices: forward, reverse, no)
strandness: forward
# Filter out reads with mapping quality below this value
min_mapping_quality: 0
# Only considier longest transcript for transcriptome mapping
use_longest_transcript: true
# Expected read length for mapping using STAR
star_sjdboverhang: 100
# Number of threads for mapping
threads_mapping: 4
# Remove duplicates for long RNA-seq before feature counting
remove_duplicates_long: false
# Input reads are paired-end
paired_end: false
# Use small RNA-seq pipeline (sequential mapping)
small_rna: true
# Remove UMI tags (leading nucleotides)
umi_tags: false
# Length of the UMI barcode
umi_length: 0
# Evaluate published biomarkers
evaluate_features_preprocess_methods: []
# Differential expression method
diffexp_method: deseq2
# Count multi-mapping reads
count_multimap_reads: true
# Count overlapping features
count_overlapping_features: true
</code></pre>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../quality_control/" class="btn btn-neutral float-right" title="Quality Control">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../cluster_configuration/" class="btn btn-neutral" title="Run on a cluster"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
          <a href="https://github.com/lulab/exSeek/" class="fa fa-github" style="float: left; color: #fcfcfc"> GitHub</a>
      
      
        <span><a href="../cluster_configuration/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../quality_control/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>

</body>
</html>
